src/ipc/ipc4/handler.c

```diff
@@ -880,8 +880,6 @@ static int ipc4_module_process_dx(struct ipc4_message_request *ipc4)
 #endif
 		/* do platform specific suspending */
 		platform_context_save(sof_get());
-
-		ipc_get()->pm_prepare_D3 = 1;
 	}
```

Comment:
Now we can do this like this:
```diff
-#if defined(CONFIG_PM)
-		ipc_get()->task_mask |= IPC_TASK_POWERDOWN;
-#endif
		/* do platform specific suspending */
		platform_context_save(sof_get());

-		ipc_get()->pm_prepare_D3 = 1;
+		ipc_get()->task_mask |= IPC_TASK_POWERDOWN;
```

and remove this from _**src\platform\intel\cavs\platform.c**_

```diff
int platform_context_save(struct sof *sof)
{
-#if CONFIG_IPC_MAJOR_4
-	ipc_get()->task_mask |= IPC_TASK_POWERDOWN;
-#endif
-
#if CONFIG_CAVS_IMR_D3_PERSISTENT
```